crates/biome_json_analyze/src/lint/nursery/no_duplicate_dependencies.rs

```diff
@@ -0,0 +1,231 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_json_syntax::{JsonRoot, JsonSyntaxKind, JsonSyntaxToken};
+use biome_rowan::{AstNode, AstSeparatedList};
+use biome_rule_options::no_duplicate_dependencies::NoDuplicateDependenciesOptions;
+use rustc_hash::FxHashMap;
+
+declare_lint_rule! {
+    /// Disallow any dependency from being specified more than once (e.g. in `dependencies` and `devDependencies`)
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```json,expect_diagnostic
+    /// {
+    ///     "dependencies": {
+    ///         "foo": "1.0.0"
+    ///     },
+    ///     "devDependencies": {
+    ///         "foo": "1.0.0"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```json
+    /// {
+    ///     "dependencies": {
+    ///         "foo": "1.0.0"
+    ///     }
+    /// }
+    /// ```
+    ///
+    pub NoDuplicateDependencies {
+        version: "next",
+        name: "noDuplicateDependencies",
+        language: "json",
+        recommended: false,
+        sources: &[RuleSource::EslintPackageJson("unique-dependencies").same(), RuleSource::EslintPackageJsonDependencies("duplicate-dependencies").same()],
+    }
+}
+
+const PACKAGE_JSON: &str = "package.json";
+
+// dependencies <-> devDependencies / optionalDependencies / peerDependencies
+// peerDependencies <-> optionalDependencies
+const UNIQUE_PROPERTY_KEYS: [(&str, &[&str]); 2] = [
+    (
+        "dependencies",
+        &[
+            "devDependencies",
+            "optionalDependencies",
+            "peerDependencies",
+        ],
+    ),
+    ("peerDependencies", &["optionalDependencies"]),
+];
+
+const DUPLICATE_PROPERTY_KEYS: &[&str; 7] = &[
+    "bundledDependencies",
+    "bundleDependencies",
+    "dependencies",
+    "devDependencies",
+    "optionalDependencies",
+    "overrides",
```

Comment:
I don't think `overrides` belongs in this list. The semantics for `overrides` are different from the other objects, and there can be good reasons for having a dependency listed under both `dependencies` and `overrides`. See the example under https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

Reply:
I think it should, so did one of the plugins. `overrides` is not checked against other dependency groups, only checked for duplicates within it's own group.

Reply:
Ah fair, in that case I agree it makes sense to include 👍